UB in mbsys_reson7k.c

```
mbsys_reson7k.c: In function ‘mbsys_reson7k_alloc’:
mbsys_reson7k.c:988:36: warning: iteration 6 invokes undefined behavior [-Waggressive-loop-optimizations]
  988 |     calibratedsnippet->reserved[i] = 0;
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
mbsys_reson7k.c:987:3: note: within this loop
  987 |   for (int i = 0; i < 28; i++)
      |   ^~~
```

Environment: debian testing

```
gcc --version
gcc (Debian 9.2.1-8) 9.2.1 20190909
```

build:

```
./autogen.sh
CFLAGS='-g -O2 -Wno-format -Wno-stringop-truncation -Wno-stringop-overflow -Wall -Wextra -Wno-sign-compare -Wno-maybe-uninitialized -Wno-format-overflow' ./configure --enable-static --disable-shared --enable-test --prefix=`pwd`/inst
make clean -j 50
make -j 50 check
```

The code:

```c
  /* Reson 7k calibrated snippet (record 7058) */
  s7kr_calibratedsnippet *calibratedsnippet = &store->calibratedsnippet;
  mbsys_reson7k_zero7kheader(verbose, &calibratedsnippet->header, error);
  calibratedsnippet->serial_number = 0;
  calibratedsnippet->ping_number = 0;
  calibratedsnippet->multi_ping = 0;
  calibratedsnippet->number_beams = 0;
  calibratedsnippet->error_flag = 0;
  calibratedsnippet->control_flags = 0;
  for (int i = 0; i < 28; i++)
    calibratedsnippet->reserved[i] = 0;   //. <-------- Issue is here
  for (int i = 0; i < MBSYS_RESON7K_MAX_BEAMS; i++) {
    s7kr_calibratedsnippettimeseries *calibratedsnippettimeseries =
        &calibratedsnippet->calibratedsnippettimeseries[i];
    calibratedsnippettimeseries->beam_number = 0;
    calibratedsnippettimeseries->begin_sample = 0;
    calibratedsnippettimeseries->detect_sample = 0;
    calibratedsnippettimeseries->end_sample = 0;
    calibratedsnippettimeseries->nalloc = 0;
    calibratedsnippettimeseries->amplitude = NULL;
  }
```

Whoops.... `s7kr_calibratedsnippet` in mbsys_reson7k.h.  28 is a lot more than 6.

```
  unsigned int reserved[6];  // Reserved for future use
```

--- a/src/mbio/mbsys_reson7k.c
+++ b/src/mbio/mbsys_reson7k.c
@@ -984,7 +984,7 @@ int mbsys_reson7k_alloc(int verbose, void *mbio_ptr, void **store_ptr, int *erro
   calibratedsnippet->number_beams = 0;
   calibratedsnippet->error_flag = 0;
   calibratedsnippet->control_flags = 0;
-  for (int i = 0; i < 28; i++)
+  for (int i = 0; i < 6; i++)
     calibratedsnippet->reserved[i] = 0;
   for (int i = 0; i < MBSYS_RESON7K_MAX_BEAMS; i++) {
     s7kr_calibratedsnippettimeseries *calibratedsnippettimeseries =

